x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculator.java

```diff
@@ -70,12 +73,56 @@ private static OptionalLong allowedBytesForMl(String nodeBytes, String jvmBytes,
         return OptionalLong.of(allowedBytesForMl(machineMemory, jvmMemory, maxMemoryPercent, useAuto));
     }
 
-    public static int modelMemoryPercent(long machineMemory, long jvmSize, int maxMemoryPercent, boolean useAuto) {
+    public static long calculateApproxNecessaryNodeSize(long nativeMachineMemory, Long jvmSize, int maxMemoryPercent, boolean useAuto) {
+        if (nativeMachineMemory == 0) {
+            return 0;
+        }
         if (useAuto) {
-            // It is conceivable that there is a machine smaller than 200MB.
-            // If the administrator wants to use the auto configuration, the node should be larger.
+            // TODO utilize official ergonomic JVM size calculations when available.
+            final long unboxedJvm;
+            // See dynamicallyCalculateJvm the following JVM calculations are arithmetic inverses of JVM calculation
+            //
+            // Example: For < 2GB node, the JVM is 0.4 * total_node_size. This means, the rest is 0.6 the node size.
+            // So, the `nativeAndOverhead` is == 0.6 * total_node_size => total_node_size = (nativeAndOverHead / 0.6)
+            // Consequently jvmSize = (nativeAndOverHead / 0.6)*0.4 = nativeAndOverHead * 2/3
+            if (jvmSize == null) {
+                long nativeAndOverhead = nativeMachineMemory + OS_OVERHEAD;
+                if (nativeAndOverhead < (ByteSizeValue.ofGb(2).getBytes() * 0.60)) {
+                    unboxedJvm = (long) Math.ceil(nativeAndOverhead * (2.0 / 3.0));
+                } else if (nativeAndOverhead < (ByteSizeValue.ofGb(8).getBytes() * 0.75)) {
+                    unboxedJvm = (long) Math.ceil(nativeAndOverhead / 3.0);
+                } else {
+                    unboxedJvm = ByteSizeValue.ofGb(2).getBytes();
+                }
+            } else {
+                unboxedJvm = jvmSize;
+            }
+            // Due to rounding errors between calculating, we account for percentage inaccuracies so that our approximate node size
+            // is always SLIGHTLY too large (or exactly correct).
+            int modelMemoryPercent = modelMemoryPercent(
+                nativeMachineMemory + unboxedJvm + OS_OVERHEAD,
+                unboxedJvm,
+                maxMemoryPercent,
+                true) - AUTO_MEMORY_ERROR_RATE;
```

Comment:
I am not sure that having the `1` in a constant called `AUTO_MEMORY_ERROR_RATE` makes this clearer.  It's not really a rate.  I don't see how the rounding errors in calculating the percentage should ever result in the rounded percentage being off by more than 1.  The problem really is that the `modelMemoryPercent` method does a `Math.ceil`, when for this particular case we wanted a `Math.floor`.  So I would just hardcode 1 here and change the comment to say it's because `modelMemoryPercent` uses `Math.ceil` and we wanted `Math.floor`.  The alternatives would be:

1. Add an extra argument to `modelMemoryPercent` telling it whether to round up or down.
2. Make `modelMemoryPercent` return a `double`, and let the caller choose whether to round up or down.

Since this PR is quite urgent I'm happy to go with the low-churn solution of just subtracting 1 here and changing the comment.